portfolio/src/main/java/com/google/sps/agents/WorkoutAgent.java

```
@@ -0,0 +1,167 @@
+   */
+  private void setWorkoutFindDisplay() throws IOException {
+
+    // Removing white space so search URL does not have spaces
+    workoutLength = workoutLength.replaceAll("\\s", "");
+    workoutType = workoutType.replaceAll("\\s", "");
+    youtubeChannel = youtubeChannel.replaceAll("\\s", "");
+
+    // Make API call to WorkoutUtils to get json object of videos
+    JSONObject json =
+        WorkoutUtils.getJSONObject(workoutLength, workoutType, youtubeChannel, numVideosDisplayed);
+    JSONArray videos = json.getJSONArray("items");
+
+    List<Video> videoList = new ArrayList<>();
+
+    for (int i = 0; i < videos.length(); i++) {
+      String videoString = new Gson().toJson(videos.get(i));
+      setVideoParameters(videoString);
+      video = new Video(channelTitle, title, description, thumbnail, videoId);
+      videoList.add(video);
+    }
+
+    display = new Gson().toJson(videoList);
+  }
+
+  private void setVideoParameters(String videoString) {
+    JSONObject videoJSONObject = new JSONObject(videoString).getJSONObject("map");
+    JSONObject id = videoJSONObject.getJSONObject("id").getJSONObject("map");
+    videoId = new Gson().toJson(id.get("videoId"));
+    JSONObject snippet = videoJSONObject.getJSONObject("snippet").getJSONObject("map");
+    description = new Gson().toJson(snippet.get("description"));
+    title = new Gson().toJson(snippet.get("title"));
+    channelTitle = new Gson().toJson(snippet.get("channelTitle"));
+    JSONObject thumbnailJSONObject =
+        snippet.getJSONObject("thumbnails").getJSONObject("map").getJSONObject("medium");
+    JSONObject thumbnailURL = thumbnailJSONObject.getJSONObject("map");
+    thumbnail = new Gson().toJson(thumbnailURL.get("url"));
+  }
+
+  /**
+   * TODO: Private workoutSchedule method
+   *
+   * @param parameters parameter Map from Dialogflow
+   */
+  private void workoutSchedule(Map<String, Value> parameters) {
+    return;
```

you might want to throw an unimplemented error here to avoid any sneaky bugs. this may be a big enough aspect thought that you're perfectly ok with quiet nothingness for the time being.